Added `hive-hooks` component for generic path and table location modification. (#76)
ExpediaGroup/apiary-extensions: hive-hooks/README.md

# Hive Hooks
This module implements Hive metastore filter hooks for use within the Apiary project.

## Path Conversion Hook
This module implements a Hive client-side metastore filter in the class `ApiaryMetastoreFilter`.

This path conversion hook allows one to arbitrarily alter partition and table locations which match regular expressions
for a given table without altering the underlying data. An example use would be introducing something like [Alluxio](https://www.alluxio.io/) as a caching 

> layer for data storage - in this case the filter would change the scheme in the file locations from `s3` to `alluxio`   so clients would read from the cache to improve read speeds. 

Reviewer: ```suggestion
layer for data storage - in this case the filter would change the scheme in the file locations from `s3` to `alluxio` so clients would read from the cache to improve read speeds. 
```

   set a `hive-site.xml` property named `apiary.path.replacement.enabled` to a truthy value ("true", "True", "yes", etc).
   ```
    <property>
        <name>apiary.path.replacement.enabled</name>
        <value>true</value>
    </property>
   ```
4. In order to properly set a regex and replacer pattern, you must add the following properties for regex, value and (optionally)
   capture groups. This should be set with a suffix qualifier to tie all of them together, like so:
    ```
    <property>
        <name>apiary.path.replacement.regex.<MY_FOO_REPLACER></name>
        <value>(foo).(baz)</value>
    </property>
    <property>
        <name>apiary.path.replacement.value.<MY_FOO_REPLACER></name>
        <value>bar</value>
    </property>
   <property>
       <name>apiary.path.replacement.capturegroups.<MY_FOO_REPLACER></name>
       <value>1,2</value>
   </property>
    ```
   **NOTE**: Each defined key-value pair of these regex and value matchers must have a matching pair or else the hook will ignore it.
   
   For example if one wanted to convert any path starting with a s3 protocol and containing the `us-west-1` region

>    to instead point at an Alluxio cluster endpoint, they can configure the hook like so:

Reviewer: ```suggestion
   to instead point at an Alluxio cluster endpoint, the hook would be configured like so:
```
